.clang-tidy

```
@@ -0,0 +1,34 @@
+Checks: >
+  -*,
+  bugprone-* ,
+  performance-* ,
+  readability-* ,
+  modernize-* ,
+  cppcoreguidelines-* ,
+  portability-* ,
+  misc-unused-* ,
+  hicpp-* ,
+  clang-analyzer-* ,
+  -modernize-use-trailing-return-type
+
+WarningsAsErrors: >
+  bugprone-* ,
+  performance-* ,
+  clang-analyzer-*
+
+HeaderFilterRegex: 'src/.*'
```

Should this not be:

```suggestion
HeaderFilterRegex: 'src/.*\.h'?
```

Else it would match any path. Or do we want that?

---

does not matter, clang is smart enough to see if its a header or not